### ISSUE TITLE
Toolbar with widget.Label makes the ToolbarAction buttons higher

### ISSUE THREAD

**Issue by marmiha** [fyne-io/fyne#2257]
### Describe the bug:
Adding a widget.NewLabel() inside a widget.NewToolbar() makes the widget.NewToolbarAction() a little bit higher.

### To Reproduce:
Steps to reproduce the behaviour:
1. Make a custom struct inheriting widget.ToolbarItem

```
type toolbarLabel struct {
	*widget.Label
}

func NewToolbarLabel(label string) widget.ToolbarItem {
	l := widget.NewLabelWithStyle(label, fyne.TextAlignCenter ,fyne.TextStyle{Bold: true})
	l.MinSize()
	return &toolbarLabel{l}
}

func (t *toolbarLabel) ToolbarObject() fyne.CanvasObject {
	return t.Label
}
```

2. Compare the height of the buttons between the two toolbars. This can also be replicated if a Label is inside the HBox (acts the same as in Toolbar).
```
func MainScreen() fyne.CanvasObject {
	
	// Menu
	toolbar_hbox := container.NewHBox(
		widget.NewToolbarAction(theme.FolderIcon(), func() {
			fmt.Println("profile settings clicked in toolbar")
		}).ToolbarObject(),

		widget.NewToolbarSpacer().ToolbarObject(),
		widget.NewToolbarAction(theme.FolderIcon(), func() {
			fmt.Println("log out clicked in toolbar")
		}).ToolbarObject(),
	)

	toolbar_tb := widget.NewToolbar(
		widget.NewToolbarAction(theme.FolderIcon(), func() {
			fmt.Println("profile settings clicked in toolbar")
		}),
		element.NewToolbarLabel("marmiha"),
		widget.NewToolbarSpacer(),
		widget.NewToolbarAction(theme.FolderIcon(), func() {
			fmt.Println("log out clicked in toolbar")
		}),
	)


	return container.NewBorder(toolbar_tb, toolbar_hbox, nil, nil)
}

```
### Screenshots:
![image](https://user-images.githubusercontent.com/36703826/119009142-2068c400-b993-11eb-9cd1-fdc557770bfa.png)
![image](https://user-images.githubusercontent.com/36703826/119009177-26f73b80-b993-11eb-9cd9-7c7c9e245e08.png)

### Device (please complete the following information):
 - **OS:** Windows
 - **Version:** 10
 - **Go version:** go 1.16.3 windows/amd64
 - **Fyne version:**  v2.0.3

**Comment by andydotxyz** [comment]
I think we may have broken this with the 2.0 changes to how widgets are padded.

**Comment by andydotxyz** [comment]
Fix is on `develop` branch now, will be in 2.1 release.
